.github/workflows/test-python.yml

```diff
@@ -146,3 +146,73 @@ jobs:
             pip uninstall "$d" -y
             python -c 'import polars'
           done
+
+  python-narwhals-testsuite:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.13']
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set up Graphviz
+        uses: ts-graphviz/setup-graphviz@v2
+
+      - name: Create virtual environment
+        env:
+          BIN: ${{ matrix.os == 'windows-latest' && 'Scripts' || 'bin' }}
+        run: |
+          python -m venv .venv
+          echo "$GITHUB_WORKSPACE/py-polars/.venv/$BIN" >> $GITHUB_PATH
+          echo "VIRTUAL_ENV=$GITHUB_WORKSPACE/py-polars/.venv" >> $GITHUB_ENV
+          echo "VENV=$GITHUB_WORKSPACE/py-polars/.venv" >> $GITHUB_ENV
+
+      - name: Install Python dependencies
+        run: |
+          pip install uv
+          uv pip install --compile-bytecode maturin pytest pytest-env --verbose
+      - name: Set up Rust
+        run: rustup show
+
+      - name: Cache Rust
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: py-polars
+          save-if: ${{ github.ref_name == 'main' }}
+
+      - name: Install Polars
+        run: |
+          uv pip install --no-deps -e .
+          maturin develop --manifest-path runtime/Cargo.toml
+
+      - name: Clone & install Narwhals 
+        run: |
+          git clone https://github.com/narwhals-dev/narwhals.git --depth=1
+          cd narwhals
+          uv pip install -e ".[pandas,pyarrow]" --group tests
```

Comment:
Currently we can't run all the tests without having pandas and pyarrow installed.
We have a issue open for quite some time on this topic (https://github.com/narwhals-dev/narwhals/issues/1726) but never finished working on it. This might be a good motivation to prioritize it